src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -214,3 +214,18 @@ def random_string(length=16, force_lower=False, digits_only=False):
     if not digits_only:
         choice_set += ascii_lowercase if force_lower else ascii_letters
     return ''.join([choice(choice_set) for _ in range(length)])
+
+
+def hash_string(value, length=16, force_lower=False):
```

Comment:
The original method this was replacing allowed an arbitrary length, so I wanted to ensure that this functionality was preserved in the replacement while still being deterministic.